src/web/components/AdSlot.tsx

```diff
@@ -87,114 +127,210 @@ const mobileStickyAdStyles = css`
-    // refresh = true,
-    // outOfPage = false,
-    optId,
-    optClassNames,
-    localStyles,
-    isSticky,
-}) => {
-    // Will export `getOptionalProps` as a function if/when needed - Pascal.
-    // const getOptionalProps = (): object => ({
-    //     ...(showLabel && { 'data-label': true }),
-    //     ...(refresh && { 'data-refresh': true }),
-    //     ...(outOfPage && { 'data-out-of-page': true }),
-    // });
-
-    const postionStyles = isSticky
-        ? css`
-              position: sticky;
-              top: 0;
-          `
-        : css`
-              position: relative;
-          `;
-
-    const sizeMappings = makeInternalSizeMappings(sizeMapping);
-    return (
-        <div
-            id={`dfp-ad--${optId || name}`}
-            className={`${makeClassNames(
-                name,
-                adTypes,
-                optClassNames || [],
-            )} ${postionStyles} ${localStyles} ${labelStyles}`}
-            data-link-name={`ad slot ${name}`}
-            data-name={name}
-            // {...getOptionalProps()}
-            // eslint-disable-next-line react/jsx-props-no-spreading
-            {...sizeMappings}
-            aria-hidden="true"
-        />
-    );
-};
-
-export const AdSlot: React.FC<{
-    asps: AdSlotParameters;
-    localStyles?: string;
-    isSticky?: boolean;
-    heightToStick?: string;
-}> = ({ asps, localStyles, isSticky, heightToStick }) => {
-    if (isSticky) {
-        return (
-            <div
-                className={css`
-                    position: static;
-                    height: ${heightToStick || '100%'};
-                `}
-            >
-                <AdSlotCore {...asps} isSticky={true} />
-            </div>
-        );
+export const AdSlot: React.FC<Props> = ({ position, display }) => {
+    switch (position) {
+        case 'right':
+            switch (display) {
+                case Display.Immersive:
+                case Display.Showcase: {
+                    return (
+                        <div
+                            id="dfp-ad--right"
+                            className={cx(
+                                'js-ad-slot',
+                                'ad-slot',
+                                'ad-slot--right',
+                                'ad-slot--mpu-banner-ad',
+                                'ad-slot--rendered',
+                                'js-sticky-mpu',
+                                css`
+                                    position: sticky;
+                                    top: 0;
+                                `,
+                                labelStyles,
+                            )}
+                            data-link-name="ad slot right"
+                            data-name="right"
+                            // mark: 01303e88-ef1f-462d-9b6e-242419435cec
+                            data-mobile={`${Size.outOfPage}|${Size.empty}|${Size.mpu}|${Size.googleCard}|${Size.halfPage}|fluid`}
```

Comment:
For consistency:

```suggestion
                            data-mobile={`${Size.outOfPage}|${Size.empty}|${Size.mpu}|${Size.googleCard}|${Size.halfPage}|${Size.fluid}`}
```

It’s also used in other locations in this file.

Reply:
I think you're right. It's hard to read these data attributes with all the variables jumbled up in there, I'm not sure though if there's a good solution here, string manipulations are rarely easy to read! Maybe this code shouldn't be simplified even further and just written out as it will end up in the browser?

So instead of

```html
      data-name="right"
      // mark: 01303e88-ef1f-462d-9b6e-242419435cec
      data-mobile={`${Size.outOfPage}|${Size.empty}|${Size.mpu}|${Size.googleCard}|${Size.halfPage}|fluid`}
      aria-hidden="true"
```

we have:


```html
      data-name="right"
      // mark: 01303e88-ef1f-462d-9b6e-242419435cec
      data-mobile="1,1|2,2|300,250|300,274|300,600|fluid"
      aria-hidden="true"
```

I quite like this simplicity; it's certainly clear and simple to read. The disadvantage though is that you lose the intrinsic naming of the different sizes. (You could also say that it's nice to change the code in one place and have that propogate but I text editors today make that kind of work trivial.)

What do you think? I'm happy to defer to your opinion here. You probably have a better idea on the importance of these size names so if you think they're worth keeping I'll try to tidy things up to be more consistent. But if you think they can be lost I'll par things down to pure html!

Reply:
another option that could be more readable could be something like this:
```
data-mobile={[
    `${Size.outOfPage}`
    `${Size.empty}`
    `${Size.mpu}`
    `${Size.googleCard}`
    `${Size.halfPage}`
    `fluid`
  ].join("|")
}
```

Reply:
This is much nicer! Now it's an array, prettier is able do its thing